staging/src/k8s.io/component-base/metrics/prometheus/restclient/metrics.go

```diff
@@ -103,6 +104,16 @@ var (
 			},
 		},
 	)
+
+	execPluginCalls = k8smetrics.NewCounterVec(
+		&k8smetrics.CounterOpts{
+			Name: "rest_client_exec_plugin_call_total",
+			Help: "Number of calls to an exec plugin, partitioned by exit code. A special exit code of -1 " +
```

Comment:
I wanted to document a contract for what the exec authenticator does when it encounters exec plugin errors that it doesn't know how to process (e.g., the go stdlib failed to copy IO streams (very unlikely), the user does not have permission to exec the plugin). I figured this was a good enough place to document it since this help text will most likely be displayed to users in some fashion?

I think we are safe with using negative numbers as special values, since POSIX uses the low-order 8 (unsigned) bits of an `int` return value to indicate the status, and Windows treats return values as unsigned integers. See https://en.wikipedia.org/wiki/Exit_status.

Reply:
I think we do this elsewhere too actually.

@liggitt may know?

Reply:
Another question here - do we need to document this metric publicly in any k/website (or elsewhere) docs? I am not familiar with how kube metrics are documented.

Reply:
> I think we are safe with using negative numbers as special values, since POSIX uses the low-order 8 (unsigned) bits of an int return value to indicate the status, and Windows treats return values as unsigned integers. See https://en.wikipedia.org/wiki/Exit_status.
>
> ...
> I think we do this elsewhere too actually.
> @liggitt may know?

umm.... I ... don't know

> do we need to document this metric publicly in any k/website (or elsewhere) docs?

In the KEP is a good starting point (https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/541-external-credential-providers/README.md#metrics). including in the [user-doc](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins) as well would make sense to me